File: bin/remove_dup_seqs.pl
use warnings;
use strict;
use feature 'say';


my $seq_hash = {};

my @all_lines = ();
my $count = 0;
open(IN,$ARGV[0]);
while(<IN>) {
  my $line = $_;
  chomp($line);
  if($line =~ /\>/) {
    if($count) {
      $count++;
      $all_lines[$count] = $line;
      $count++;
    } else {
      $all_lines[$count] = $line;
      $count++;
    }
  } else {
    if($all_lines[$count]) {
      $all_lines[$count] .= $line;
    } else {
      $all_lines[$count] = $line;
    }
  }

}
close IN;

for(my $i=0; $i<scalar(@all_lines); $i+= 2) {
  my $header = $all_lines[$i];
  my $seq = $all_lines[$i+1];
  unless(length($seq) >= 100) {
    next;
  }
  $seq_hash->{$seq} = $header;
}

foreach my $key (keys(%{$seq_hash})) {
  say $seq_hash->{$key};
  say $key;
}

exit;

File: bin/remove_dup_seqs.py
import sys
import os

input_file = sys.argv[1]
if len(sys.argv) != 2 or not os.path.exists(input_file):
    print("Usage: python remove_dup_seqs.py input_file\nPerhaps the input_file was missing")
    sys.exit(1)

seq_hash = {}

with open(input_file, 'r') as infile:
    header = None
    sequence = ""

    for line in infile:
        line = line.strip()

        if not line:
            continue  # Skip empty lines

        if line.startswith('>'):
            # If a header line is encountered, store the previous sequence
            if header is not None and sequence:
                seq_hash[sequence] = header

            # Reset header and start a new sequence
            header = line
            sequence = ""
        else:
            sequence += line

    # Process the last sequence in the file
    if header is not None and sequence:
        seq_hash[sequence] = header

# Print unique sequences along with their headers
for sequence, header in seq_hash.items():
    print(header)
    print(sequence)

